hyperhtml.js

```diff
@@ -1,56 +1,43 @@
-var hyperHTML = (function () {'use strict';
+var hyperHTML = (function (document) {'use strict';
 
   /*! (c) 2017 Andrea Giammarchi @WebReflection, (MIT) */
 
-  // hyperHTML \o/
-  //
-  // var render = hyperHTML.bind(document.body);
-  // setInterval(() => render`
-  //  <h1>⚡️ hyperHTML ⚡️</h1>
-  //  <p>
-  //    ${(new Date).toLocaleString()}
-  //  </p>
-  // `, 1000);
-  function hyperHTML(statics) {
-    return  EXPANDO in this &&
-            this[EXPANDO].s === statics ?
-              update.apply(this, arguments) :
-              upgrade.apply(this, arguments);
-  }
-
-  // Specially useful when you have content rendered
-  // on the server and you don't want to lose it
-  // at the first time hyperHTML passes over it.
-  // To be used instead of hyperHTML.bind(node)
-  // but only if the content of the node is the equivalent
-  // of the used template literal.
-  //
-  // <body><div adopted=""> adopt me </div></body>
-  //
-  // var render = hyperHTML.adopt(document.body);
-  // render`<div adopted="${true}"> ${'new text'} </div>`;
+  // ---------------------------------------------
+  // hyperHTML Public API
+  // ---------------------------------------------
+
+  // The document must be swap-able at runtime.
+  // Needed by both basicHTML and nativeHTML
+  hyperHTML.document = document;
+
+  // hyperHTML.bind(el) ⚡️
+  function hyperHTML(template) {
+    var hyper = hypers.get(this);
+    if (
+      !hyper ||
+      hyper.template !== (FF ? unique(template) : template)
+    ) {
+      var fragment = upgrade.apply(this, arguments);
+      if (notAdopting) {
```

Comment:
I rather not touch arguments at all and `notAdopting` is private in the scope (just clarifying since you said it's global)

You can see `notAdopting` as the only meaningful _state_ in `hyperHTML`